src/units/unit.cpp

```diff
@@ -1472,8 +1481,7 @@ static bool matches_ability_filter(const config & cfg, const std::string& tag_na
 		return false;
 
 	if(!filter["value"].empty()){
-		bool has_other_key = (!cfg["add"].empty() || !cfg["sub"].empty() || !cfg["multiply"].empty() || !cfg["divide"].empty());
-		if(!has_other_key){
+		if(!has_value_changing_modifier(cfg)){
```

Comment:
Looking at the conditional block contained by this, I don't understand the `else` block. If there *is* a value-changing modifier, it checks whether the unchanged value matches.

Reply:
when has_value_changing_modifier is true, then if not has_attribute_value, return false because default value of abilities will be changed, and absence of attribut eqals. It's only when has_value_changing_modifier is false what default value can be checked if not has_attribute("value")(in line 1484 has_value_changing_modifier() is false)